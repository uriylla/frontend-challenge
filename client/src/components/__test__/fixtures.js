export const clusters = [
  {
    id: '30150817',
    time: '2019-12-01T17:36:05Z',
    story_count: 67028,
    earliest_story: '2019-11-22T21:50:32Z',
    latest_story: '2019-12-12T23:50:45Z',
    location: {
      country: 'TR',
    },
    representative_story: {
      id: '47266523',
      title:
        'ميركل تعزى رئيس وزراء ألبانبا فى وفاة زوجة نجله فى الزلزال - اخبار',
      body:
        'أعربت المستشارة الألمانية أنجيلا    ميركل   عن تعاطفها الشديد لذوي ضحايا ال   زلزال   الذي ضرب ألبانيا وللحكومة الألبانية.\nوكتبت    ميركل   اليوم الأحد في برقية تعزية لرئيس الوزراء الألباني إيدي راما: “اعلم الألم الذي اضطررت أنت وعائلتك لمعايشته”.\nوبحسب تقارير، توفيت شريكة حياة نجل رئيس الوزراء الألباني خلال ال   زلزال  .\nوأضافت المستشارة: ”   ألمانيا   والاتحاد الأوروبي سيقدمان دعمهما لكم في هذا الوقت العصيب”.\nيذكر أن 50 شخصًا لقوا حتفهم إثر ال   زلزال   الذي ضرب ألبانيا يوم الثلاثاء الماضي والذي يعد الأقوى هناك منذ عقود. وتوقفت قوات الإنقاذ عن البحث عن ضحايا أمس السبت. وأصيب إثر ال   زلزال   أكثر من 650 شخصًا. وبحسب بيانات الحكومة الألبانية، ضرب    زلزال   بقوة 4ر6 ريختر نحو ألفي مبنى. ووفقا لمعهد علوم الأرض في تيرانا، وقع مركز ال   زلزال   على بعد عشرة كيلومترات شمالي ميناء دوريس الغربي وعلى بعد 30 كيلومترا غربي العاصمة الألبانية تيرانا.\n\n  يمكن قراءة الخبر من المصدر من هنا',
      sentences: [
        'German Chancellor Angela Merkel expressed her deep sympathy for the victims of the earthquake that hit Albania and the Albanian government.',
        'According to reports, a partner of the life of a Prime Minister of Albania died during the earthquake.',
      ],
      link: 'https://www.libyaakhbar.com/news/1080980.html',
      hashtags: [
        '#Earthquake',
        '#Albania',
        '#Albanians',
        '#PrimeMinisterOfTheUnitedKingdom',
        '#Tirana',
        '#ChancellorOfGermany',
        '#AngelaMerkel',
        '#Telegraphy',
        '#Rama',
        '#Pain',
        '#Germany',
        '#EuropeanUnion',
        '#Earth',
        '#Port',
      ],
      media: [
        {
          type: 'image',
          url:
            'https://www.libyaakhbar.com/wp-content/uploads/2019/12/637107994911337082_131224338148_433_LTH.jpg',
        },
      ],
      source: {
        id: '16411',
        name: 'Libya Akhbar',
        home_page_url: 'https://www.libyaakhbar.com/',
      },
      author: {
        id: '496316',
        name: 'ايوان ليبيا',
      },
    },
  },
  {
    id: '29783373',
    time: '2019-12-04T08:41:40Z',
    story_count: 49262,
    earliest_story: '2019-11-20T11:20:50Z',
    latest_story: '2019-12-08T12:38:04Z',
    location: {
      country: 'CN',
    },
    representative_story: {
      id: '47867739',
      title: 'Dopo una chemioterapia',
      body:
        'Mio padre, operato di tumore al colon a inizio estate, ha poi fatto cicli di chemioterapia e altri farmaci a bersaglio molecolare. Ora sta bene e le terapie sono da poco finite. Ci chiediamo se sia il caso di fargli fare il vaccino antinfluenzale: ci sono controindicazioni? Dobbiamo chiedere all’oncologo?\n\n\n\n\t\n\n\nRisponde Giordano Beretta, presidente eletto dell’Associazione Italiana di Oncologia Medica (Aiom) e responsabile dell’Oncologia Medica di Humanitas Gavazzeni Bergamo\n\n\n\n\n\t\n\t\t\n\t\t\t\n\n\t\t\t\n\t\t\n\t\n\n\nIl vaccino antinfluenzale Ã¨ indicato per tutti i soggetti che desiderano evitare l’influenza e che non abbiano specifiche controindicazioni. Ã offerto gratuitamente a chi Ã¨ a maggior rischio di complicanze: bambini, persone d’etÃ  superiore a 65 anni e soggetti che soffrono di patologie croniche, tra i quali i pazienti oncologici. E la gratuitÃ  si estende a tutti gli operatori sanitari e, con alcune differenze nei diversi Sistemi Sanitari Regionali, ai familiari delle persone a rischio. Il costo di una dose di vaccino, quando non gratuito, Ã¨ di circa 20 euro e la vaccinazione puÃ² essere effettuata presso i servizi vaccinali delle ASL o presso i medici di medicina generale. \n\n\n\n\n\n\nVaccini sicuri, pochi e poco probabili \nVeniamo ai suoi quesiti: i vaccini antinfluenzali autorizzati sono sicuri e sottoposti a controlli accurati. Gli effetti collaterali piÃ¹ comuni sono reazioni locali (lieve dolore e arrossamento nel punto di iniezione) e, meno frequenti febbre, dolori muscolari o articolari e mal di testa, generalmente di scarsa entitÃ  e che si risolvono entro uno o due giorni. Raramente si possono verificare allergie, mentre reazioni piÃ¹ gravi sono segnalate (non sempre legate al vaccino) solo in uno o due soggetti ogni milione di vaccinati. \n\n\n\n\n\n\nParlare con il medico per valutare il singolo caso\nAnche nei malati oncologici la vaccinazione Ã¨ indicata, sicura, minimamente invasiva e puÃ² essere somministrata anche contemporaneamente ad altri eventuali vaccini. Va utilizzata sia nei pazienti che hanno giÃ  concluso le cure sia in quelli in corso di trattamento attivo, per i quali si suggerisce di effettuarla dopo circa due settimane dal ciclo di terapia per consentire una migliore risposta immunitaria. Ci sono, come sempre, delle controindicazioni: in chi, ad esempio, ha avuto precedenti reazioni di tipo anafilattico ad uno dei componenti del vaccino. Oppure in presenza di una malattia acuta di medio-grave entitÃ  Ã¨ bene rinviare l’immunizzazione antinfluenzale a quando il paziente si sarÃ  rimesso. Per questo Ã¨ bene parlarne sia con l’oncologo che con il medico di famiglia, in modo da valutare al meglio il singolo caso, che potranno consigliarvi anche sulle principali norme igieniche per prevenire l’influenza che si scatena nel periodo autunnale e invernale: in particolare Ã¨ importante evitare luoghi affollati, seguire un’adeguata igiene delle mani e aerare le stanze dove si soggiorna.La vaccinazione deve solitamente essere effettuata nel periodo autunnale a partire dalla metÃ  di ottobre fino a fine dicembre.\n\n\n\n\n\n\n3 dicembre 2019 (modifica il 3 dicembre 2019 | 12:59)\n\n© RIPRODUZIONE RISERVATA',
      sentences: [
        'Ci chiediamo se sia il caso di fargli fare il vaccino antinfluenzale: ci sono controindicazioni?',
        'Risponde Giordano Beretta, presidente eletto dell’Associazione Italiana di Oncologia Medica (Aiom) e responsabile dell’Oncologia Medica di Humanitas Gavazzeni Bergamo\n\n\n\n\n\t\n\t\t\n\t\t\t\n\n\t\t\t\n\t\t\n\t\n\n\nIl vaccino antinfluenzale Ã¨ indicato per tutti i soggetti che desiderano evitare l’influenza e che non abbiano specifiche controindicazioni.',
      ],
      link:
        'https://news.google.com/__i/rss/rd/articles/CBMinwFodHRwczovL3d3dy5jb3JyaWVyZS5pdC9zYWx1dGUvc3BvcnRlbGxvX2NhbmNyby8xOV9kaWNlbWJyZV8wMy9kb3BvLWNoZW1pb3RlcmFwaWEtc2ktcHVvLWZhcmUtdmFjY2luby1hbnRpbmZsdWVuemFsZS1hOWY5ZWUzNi0wZDI1LTExZWEtOGZiNS0wMDg3NDk4MjRjOGQuc2h0bWzSAaIBaHR0cHM6Ly93d3cuY29ycmllcmUuaXQvc2FsdXRlL3Nwb3J0ZWxsb19jYW5jcm8vMTlfZGljZW1icmVfMDMvZG9wby1jaGVtaW90ZXJhcGlhLXNpLXB1by1mYXJlLXZhY2Npbm8tYW50aW5mbHVlbnphbGUtYTlmOWVlMzYtMGQyNS0xMWVhLThmYjUtMDA4NzQ5ODI0YzhkX2FtcC5odG1s?oc=5',
      hashtags: [
        '#Oncologia',
        '#Vaccino',
        '#Vaccinazione',
        '#Chemioterapia',
        '#Dell',
        '#Singolo',
        '#2019',
        '#PersonaggiDiToLoveRu',
        '#Neoplasia',
        '#Colon',
        '#Estate',
        '#Farmaco',
        '#FabbricaDArmiPietroBeretta',
        '#Bergamo',
        '#Euro',
        '#ASL',
        '#AziendaSanitariaLocale',
        '#IMedici',
        '#MedicinaGenerale',
        '#EffettoCollaterale',
        '#Dolore',
        '#Iniezione',
        '#Febbre',
        '#Mialgia',
        '#Allergia',
        '#BAFTAAllaMiglioreAttriceProtagonista',
        '#RispostaImmunitaria',
        '#SistemaImmunitario',
        '#Famiglia',
        '#Inverno',
        '#Igiene',
      ],
      media: [
        {
          type: 'image',
          url:
            'https://images2.corriereobjects.it/methode_image/socialshare/2019/11/22/c19a7404-14f3-11ea-b557-51f830ff2b1f.jpg',
        },
      ],
      source: {
        id: '7109',
        name: 'Google News Italia (Italian)',
        home_page_url: 'http://news.google.it/',
      },
      author: {
        id: '1839577',
        name: 'Giodano Beretta',
      },
    },
  },
];
