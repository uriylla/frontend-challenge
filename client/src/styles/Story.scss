.story {
  margin: 1em;
  @media (min-width: $breakpoint-desktop) {
    overflow: scroll;
  }
}

.story__image {
  text-align: center;
  img {
    width: 100%;
    object-fit: cover;
    @media (min-width: $breakpoint-desktop) {
      max-width: 70%;
    }
  }
}

.story__header {
  text-align: center;
}

.story__hashtags {
  font-size: 0.8em;
}

.story__link {
  color: $link-color;
}
