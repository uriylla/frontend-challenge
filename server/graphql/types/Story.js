const { GraphQLObjectType, GraphQLString, GraphQLID, GraphQLList } = require('graphql');
const { sourceResolver, authorResolver, mediaResolver, linkResolver, sentencesResolver } = require('../resolvers/story');

const Source = new GraphQLObjectType({
  name: 'Source',
  fields: {
    id: {
      type: GraphQLID,
    },
    name: {
      type: GraphQLString,
    },
    home_page_url: {
      type: GraphQLString,
    }
  }
});

const Author = new GraphQLObjectType({
  name: 'Author',
  fields: {
    id: {
      type: GraphQLID,
    },
    name: {
      type: GraphQLString,
    }
  }
});

const Media = new GraphQLObjectType({
  name: 'Media',
  fields: {
    type: {
      type: GraphQLString,
    },
    url: {
      type: GraphQLString,
    }
  }
});

const Story = new GraphQLObjectType({
  name: 'Story',
  fields: {
    id: {
      type: GraphQLID,
    },
    title: {
      type: GraphQLString,
    },
    body: {
      type: GraphQLString,
    },
    published_at: {
      type: GraphQLString,
    },
    link: {
      type: GraphQLString,
      resolve: linkResolver
    },
    hashtags: {
      type: new GraphQLList(GraphQLString)
    },
    media: {
      type: new GraphQLList(Media),
      resolve: mediaResolver
    },    
    source: {
      type: Source,
      resolve: sourceResolver
    },
    author: {
      type: Author,
      resolve: authorResolver
    },
    sentences: {
      type: new GraphQLList(GraphQLString),
      resolve: sentencesResolver
    }
  }
});

module.exports = Story;