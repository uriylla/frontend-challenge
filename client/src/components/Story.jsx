import React from 'react';
import { StoryPropType } from './types';

const Story = ({story}) => {
  return (
    <div className="story">
      <img src={story.media[0] ? story.media[0].url : '/media-placeholder.png'} className="story__image"></img>
      <h4 className="story__title">{story.title}</h4>
      <div className="story__body">
      {story.sentences.map((sentence, i) => 
        <p className="story-body story__sentence" key={i}>{sentence}</p>
      )}
      </div>
      <p>Author: {story.author.name}</p>
      <a className="story__link" target="_blank" href={story.link}>Visit post on {story.source.name}</a>
      <ul className="story__hashtags">
        {story.hashtags.map(hashtag => (
          <p key={hashtag}>{hashtag}</p>
        ))}
      </ul>
    </div>
  );
};

Story.propTypes = {
  story: StoryPropType.isRequired,
};

export default Story;