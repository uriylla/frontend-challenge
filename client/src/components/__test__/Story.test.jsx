import React from 'react';
import { render } from '@testing-library/react';
import Story from '../Story';
import { clusters } from './fixtures';

describe('Story', () => {

  const renderComponent = () => render(
      <Story story={clusters[0].representative_story} />
  );

  it('should render correctly with selected item', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
}); 