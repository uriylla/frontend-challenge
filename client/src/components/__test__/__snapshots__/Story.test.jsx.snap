// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story should render correctly with selected item 1`] = `
<div>
  <div
    class="story"
  >
    <img
      class="story__image"
      src="https://www.libyaakhbar.com/wp-content/uploads/2019/12/637107994911337082_131224338148_433_LTH.jpg"
    />
    <h4
      class="story__title"
    >
      ميركل تعزى رئيس وزراء ألبانبا فى وفاة زوجة نجله فى الزلزال - اخبار
    </h4>
    <div
      class="story__body"
    >
      <p
        class="story-body story__sentence"
      >
        German Chancellor Angela Merkel expressed her deep sympathy for the victims of the earthquake that hit Albania and the Albanian government.
      </p>
      <p
        class="story-body story__sentence"
      >
        According to reports, a partner of the life of a Prime Minister of Albania died during the earthquake.
      </p>
    </div>
    <p>
      Author: 
      ايوان ليبيا
    </p>
    <a
      class="story__link"
      href="https://www.libyaakhbar.com/news/1080980.html"
      target="_blank"
    >
      Visit post on 
      Libya Akhbar
    </a>
    <ul
      class="story__hashtags"
    >
      <p>
        #Earthquake
      </p>
      <p>
        #Albania
      </p>
      <p>
        #Albanians
      </p>
      <p>
        #PrimeMinisterOfTheUnitedKingdom
      </p>
      <p>
        #Tirana
      </p>
      <p>
        #ChancellorOfGermany
      </p>
      <p>
        #AngelaMerkel
      </p>
      <p>
        #Telegraphy
      </p>
      <p>
        #Rama
      </p>
      <p>
        #Pain
      </p>
      <p>
        #Germany
      </p>
      <p>
        #EuropeanUnion
      </p>
      <p>
        #Earth
      </p>
      <p>
        #Port
      </p>
    </ul>
  </div>
</div>
`;
